xdsl/pattern_rewriter.py

```diff
@@ -91,6 +91,11 @@ class PatternRewriter(PatternRewriterListener):
     has_done_action: bool = field(default=False, init=False)
     """Has the rewriter done any action during the current match."""
 
+    def __init__(self, current_operation: Operation):
```

Comment:
You could possibly do this with a `__post_init__`

Reply:
The `self.current_operation = current_operation` seems like strictly `__init__` code to me

Reply:
I cannot actually do a `__post_init__`, because if I would use the generated `__init__`, this would require the arguments for `PatternRewriterListener` and `Builder`. But the `Builder` arguments gets constructed in the `__init__` I wrote here.